Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Cancel.hs

```diff
@@ -216,3 +232,15 @@ driverDistanceToPickup merchantId merchantOperatingCityId tripStartPos tripEndPo
           travelMode = Just Maps.CAR
         }
   return $ distRes.distance
+
+disputeCancellationDues :: (Id Person.Person, Id Merchant.Merchant) -> Flow APISuccess
+disputeCancellationDues (personId, merchantId) = do
+  person <- QP.findById personId >>= fromMaybeM (PersonNotFound personId.getId)
+  when (person.cancellationDues == 0) $ do
+    throwError $ InvalidRequest "Cancellation Due Amount is 0"
+  merchant <- CQM.findById merchantId >>= fromMaybeM (MerchantNotFound merchantId.getId)
+  let duesLimitAmount = fromIntegral merchant.cancellationFee * merchant.rideCancellationLimit
+  when (person.disputeChancesUsed >= merchant.disputeLimit || (fromIntegral person.cancellationDues) >= duesLimitAmount) $ do
+    throwError $ InvalidRequest "Customer cannot dispute cancellation dues as dispute chances limit has reached or customer's cancellation dues has reached the limit"
+  QP.updateDisputeChancesUsed personId (person.disputeChancesUsed + 1) >> QP.updateCancellationDues personId 0
```

Comment:
Even if he has multiple cancellations, this will make the dues as 0. I think one dispute should only be for one cancellation

Reply:
Yes true. Dispute means full Amount is getting paid not just for one cancellation